### PR TITLE
PRM-1506 Removed old API code

### PR DESCRIPTION
**Pull request by AndrewRButcher**
Need changes on https://github.com/hmrc/penalties/blob/main/conf/app.routes#L3 to remove query params before merging

### CODE REVIEW

File: app/controllers/CalculationController.scala [hunk 1]
```diff
@@ -16,111 +16,55 @@
...
+import java.time.LocalDate
+import java.time.temporal.ChronoUnit
+import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
 class CalculationController @Inject()(viewLPP: CalculationLPPView,
                                       viewAdd: CalculationAdditionalView,
-                                      penaltiesService: PenaltiesService,
-                                      penaltiesServiceV2: PenaltiesServiceV2,
+                                      penaltiesServiceV2: PenaltiesService,
```
Comment (LewisMitchell) [inline app/controllers/CalculationController.scala:40]: Does this still need to be named as V2?

File: app/controllers/CalculationController.scala [hunk 2]
```diff
@@ -16,111 +16,55 @@
...
 
-  def onPageLoad(penaltyId: String, isAdditional: Boolean): Action[AnyContent] = authorise.async { implicit request =>
-    logger.debug(s"[CalculationController][onPageLoad] - Making call to old endpoint")
-    getOldPenaltyData(penaltyId, isAdditional)
-  }
-
-  def onPageLoadForNewAPI(principalChargeReference: String, penaltyCategory: String): Action[AnyContent] = authorise.async { implicit request =>
-    logger.debug(s"[CalculationController][onPageLoadForNewAPI] - Making call to new endpoint")
+  def onPageLoad(principalChargeReference: String, penaltyCategory: String): Action[AnyContent] = authorise.async { implicit request =>
+    logger.debug(s"[CalculationController][onPageLoad] - Making call to new endpoint")
```
Comment (LewisMitchell) [inline app/controllers/CalculationController.scala:49]: Is this log message useful, if so, do we need to specify 'new'?

File: app/controllers/IndexController.scala [hunk 3]
```diff
@@ -16,107 +16,69 @@
...
-                .addingToSession(latestLSPCreationDate -> latestLSPCreation.get.toString,
-                  pointsThreshold -> penaltyData.lateSubmissionPenalty.map(_.summary.regimeThreshold).getOrElse(0).toString)
-            } else {
-              result
-                .removingFromSession(allKeysExcludingAgentVRN: _*)
-            }
+    penaltiesServiceV2.getPenaltyDataFromEnrolmentKey(EnrolmentKeys.constructMTDVATEnrolmentKey(request.vrn)).map {
+      _.fold(
+        errors => {
+          logger.error(s"[OtherReasonController][getPenaltyDetails] - Received error with status ${errors.status} and body ${errors.body} rendering ISE.")
```
Comment (LewisMitchell) [inline app/controllers/IndexController.scala:50]: The log doesn't match the method name
```suggestion
          logger.error(s"[IndexController][onPageLoad] - Received error with status ${errors.status} and body ${errors.body} rendering ISE.")
```

File: app/services/PenaltiesService.scala [hunk 4]
```diff
@@ -16,126 +16,84 @@
...
-  def getETMPDataFromEnrolmentKey(enrolmentKey: String)(implicit user: User[_], hc: HeaderCarrier): Future[ETMPPayload] =
-    connector.getPenaltiesData(enrolmentKey)
+  def getPenaltyDataFromEnrolmentKey(enrolmentKey: String)(implicit user: User[_], hc: HeaderCarrier): Future[GetPenaltyDetailsResponse] =
+    connector.getPenaltyDetails(enrolmentKey)
 
-  def isAnyLSPUnpaid(penaltyPoints: Seq[PenaltyPoint]): Boolean = {
-    penaltyPoints.exists(penalty => penalty.`type` == PenaltyTypeEnum.Financial &&
-      penalty.status != PointStatusEnum.Paid &&
-      !penalty.appealStatus.contains(AppealStatusEnum.Accepted) && !penalty.appealStatus.contains(AppealStatusEnum.Accepted_By_Tribunal))
+  private def findEstimatedVatInterestFromPayload(payload: GetPenaltyDetails): BigDecimal = {
```
Comment (LewisMitchell) [inline app/services/PenaltiesService.scala:35]:
```suggestion
  private def findEstimatedVATInterestFromPayload(payload: GetPenaltyDetails): BigDecimal = {
```

File: app/viewmodels/IndexPageHelper.scala [hunk 5]
```diff
@@ -162,32 +174,34 @@ class IndexPageHelper @Inject()(p: views.html.components.p,
...
+      //TODO implement functionality for VAT interest
       returnEstimatedMessageIfInterestMoreThanZero(estimatedVATInterest._1, estimatedVATInterest._2, "whatIsOwed.VATInterest"),
-
       returnEstimatedMessageIfInterestMoreThanZero(crystallisedLPPAmount, isEstimatedAmount = false, "whatIsOwed.lppAmount"),
       returnEstimatedMessageIfInterestMoreThanZero(estimatedLPPAmount, isEstimatedAmount = true, "whatIsOwed.lppAmount"),
-
       returnEstimatedMessageIfInterestMoreThanZero(penaltiesCrystalizedInterest + penaltiesEstimatedInterest, penaltiesEstimatedInterest > BigDecimal(0), "whatIsOwed.allPenalties.interest"),
       singularOrPluralAmountOfLSPs,
-      returnMessageIfOtherUnrelatedPenalties(otherUnrelatedPenalties, "whatIsOwed.otherPenalties")
+      returnMessageIfOtherUnrelatedPenalties(false, "whatIsOwed.otherPenalties")
```
Comment (LewisMitchell) [inline app/viewmodels/IndexPageHelper.scala:204]: Should we add a TODO here?

File: it/stubs/PenaltiesStub.scala [hunk 6]
```diff
@@ -18,84 +18,21 @@ package stubs
...
-            )
-          )
-        ),
-        communications = Seq.empty,
-        financial = None
-      )
-    ),
-    latePaymentPenalties = Option(Seq.empty[LatePaymentPenalty])
-  )
+  val getPenaltyDetailsUrlVatTrader: String = s"/penalties/etmp/penalties/$vrn"
```
Comment (LewisMitchell) [inline it/stubs/PenaltiesStub.scala:34]: Do we want to rename this to: 
```suggestion
  val getPenaltyDetailsUrlVATTrader: String = s"/penalties/etmp/penalties/$vrn"
```

File: test/base/TestData.scala [hunk 7]
```diff
@@ -16,10 +16,10 @@
...
 package base
 
-import models.v3.{GetPenaltyDetails, Totalisations}
-import models.v3.lsp._
-import models.v3.lpp._
-import models.v3.appealInfo._
+import models.{GetPenaltyDetails, Totalisations}
+import models.appealInfo.{AppealInformationType, AppealLevelEnum, AppealStatusEnum}
+import models.lpp.{LPPDetails, LPPDetailsMetadata, LPPPenaltyCategoryEnum, LPPPenaltyStatusEnum, LatePaymentPenalty, MainTransactionEnum}
+import models.lsp.{LSPDetails, LSPPenaltyCategoryEnum, LSPPenaltyStatusEnum, LSPSummary, LateSubmission, LateSubmissionPenalty, TaxReturnStatusEnum}
```
Comment (LewisMitchell) [inline test/base/TestData.scala:22]: Shouldn't we just use models._ and models.lpp._ etc.

File: test/controllers/CalculationControllerSpec.scala [hunk 8]
```diff
@@ -17,111 +17,32 @@
...
+import java.time.LocalDate
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
 class CalculationControllerSpec extends SpecBase with FeatureSwitching {
   val calculationView: CalculationLPPView = injector.instanceOf[CalculationLPPView]
   val calculationAdditionalView: CalculationAdditionalView = injector.instanceOf[CalculationAdditionalView]
-  val mockPenaltiesService: PenaltiesService = mock(classOf[PenaltiesService])
-  val mockPenaltiesServiceV2: PenaltiesServiceV2 = mock(classOf[PenaltiesServiceV2])
+  val mockPenaltiesServiceV2: PenaltiesService = mock(classOf[PenaltiesService])
```
Comment (LewisMitchell) [inline test/controllers/CalculationControllerSpec.scala:43]: Could we remove the V2 suffix?

File: test/controllers/IndexControllerSpec.scala [hunk 9]
```diff
@@ -23,46 +23,37 @@ import org.mockito.Mockito.{mock, reset, when}
...
-  val cardHelper: SummaryCardHelper = injector.instanceOf[SummaryCardHelper]
-  val mockPenaltiesService: PenaltiesService = mock(classOf[PenaltiesService])
-  val page2: IndexViewv2 = injector.instanceOf[IndexViewv2]
-  val indexPageHelper2: IndexPageHelperv2 = injector.instanceOf[IndexPageHelperv2]
-  val cardHelper2: SummaryCardHelperv2 = injector.instanceOf[SummaryCardHelperv2]
-  val mockPenaltiesService2: PenaltiesServicev2 = mock(classOf[PenaltiesServicev2])
+  val page2: IndexView = injector.instanceOf[IndexView]
+  val indexPageHelper2: IndexPageHelper = injector.instanceOf[IndexPageHelper]
+  val cardHelper2: SummaryCardHelper = injector.instanceOf[SummaryCardHelper]
+  val mockPenaltiesService2: PenaltiesService = mock(classOf[PenaltiesService])
```
Comment (LewisMitchell) [inline test/controllers/IndexControllerSpec.scala:41]: Could you remove the 2 suffixes?

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

File: app/connectors/PenaltiesConnector.scala [hunk 10]
```diff
@@ -17,38 +17,30 @@
...
 import scala.concurrent.{ExecutionContext, Future}
 
 class PenaltiesConnector @Inject()(httpClient: HttpClient,
                                    val appConfig: AppConfig)(implicit ec: ExecutionContext) extends FeatureSwitching {
 
 
   private val penaltiesBaseUrl: String = appConfig.penaltiesUrl
-  private def getPenaltiesDataUrl(enrolmentKey: String, isUsingNewApi: Boolean, isUsingNewFinancialApiModel: Boolean)(implicit user: User[_]): String = {
+  private def getPenaltiesDataUrl(enrolmentKey: String)(implicit user: User[_]): String = {
+    //TODO Remove Query Params
```
[the remark this answers is not in the document]
Comment (AndrewRButcher) [reply app/connectors/PenaltiesConnector.scala:34]: That was a hang over from WIP will remove

File: app/controllers/CalculationController.scala (hunk 1 above)
Comment (AndrewRButcher) [reply app/controllers/CalculationController.scala:40]: Missed this one will update

File: app/controllers/CalculationController.scala (hunk 2 above)
Comment (AndrewRButcher) [reply app/controllers/CalculationController.scala:49]: I didn't read the actual message as you say not really useful, will remove

File: app/controllers/IndexController.scala [hunk 11]
```diff
@@ -16,107 +16,69 @@
...
-                                pageHelper: IndexPageHelper,
-                                pageHelperv2: IndexPageHelperv2)(implicit ec: ExecutionContext,
-                                                                 val appConfig: AppConfig,
-                                                                 authorise: AuthPredicate,
-                                                                 errorHandler: ErrorHandler,
-                                                                 controllerComponents: MessagesControllerComponents)
+class IndexController @Inject()(view2: IndexView,
+                                penaltiesServiceV2: PenaltiesService,
+                                cardHelper2: SummaryCardHelper,
+                                pageHelperv2: IndexPageHelper)(implicit ec: ExecutionContext,
```
[the remark this answers is not in the document]
Comment (AndrewRButcher) [reply app/controllers/IndexController.scala:38]: will remove suffix

**Comment by AndrewRButcher** [comment]
PR Comments addressed

**Review by LewisMitchell — APPROVED** [review]
Approved, however will test when the backend and AT's are updated

**Pull request merged**